internal/clients/manager.go

```diff
@@ -194,9 +223,12 @@ func (c *AdminAPIClientsManager) adjustGatewayClients(discoveredAdminAPIs []admi
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// Cleanup the pending clients list.
+	c.pendingGatewayClients = c.pendingGatewayClients[:0]
```

Comment:
By resizing this slice here in a smart way unique unnecessary allocation, underlying memory will be recycled. But I wonder what about a case when for a short period of time the number of `pendingGatewayClients` will be very high (it can be arbitrarily big - `append` adds elements) and the underlying array capacity will be never reclaimed and we will waste memory. 

But to sum up I think that in practice it should not be rather a problem, I've just put my thoughts to consider